### ISSUE TITLE
Different performance on loading from saved parameters

### ISSUE THREAD

**Issue by sholtodouglas** [AntreasAntoniou/HowToTrainYourMAMLPytorch#9]
Hi,

Great code, and really informative write up on the blog! I'm trying to adapt this for a robotic imitation learning, and I'm getting great performance on past task demonstrations during training. However, on loading the saved parameters the loss is significantly worse until it's had some iterations to train - which affects using the saved parameters in new sims. 
<img width="953" alt="screen shot 2019-02-18 at 8 34 35 am" src="https://user-images.githubusercontent.com/10319863/52920245-81d9fc80-335e-11e9-8ce4-cc738ae038ba.png">

Do you have any thoughts on this? Its not the per step importance vectors, its not the task learning rates from what I can tell.

**Comment by AntreasAntoniou** [comment]
I'd have to actually look at your code repo to help on this. That being said, one of the most frequent culprits is batch normalization. Make sure the per-step summary statistics are loaded properly. If you find out what's wrong please, do let me know, so I can rectify the issue on my repo.

**Comment by AntreasAntoniou** [comment]
I forgot to mention, that I did test my system explicitly on performance consistency before/after parameter loading and it passed the test with a perfect score. This leads me to believe, that, one of my more recent commits either broke that feature or your framework has a bug somewhere.

**Comment by marcociccone** [comment]
Hi, I'm playing with the code and one reason for this behaviour might be that you are not saving the state of the outer loop optimizer (Adam), so when you restart the experiment all the running grads are re-initialized messing up the training. This is not an issue at testing time, only at training.
